ompi/util/count_disp_array.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2024   Triad National Security, LLC. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#ifndef OMPI_UTIL_COUNT_DISP_ARRAY_H
+#define OMPI_UTIL_COUNT_DISP_ARRAY_H
+
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+
+enum ompi_count_array_type {
+    OMPI_COUNT_ARRAY_TYPE_INT,
+    OMPI_COUNT_ARRAY_TYPE_SIZE_T,
+};
+
+typedef struct ompi_count_array {
+    enum ompi_count_array_type type;
+    union {
+        const int *int_array;
+        const size_t *size_t_array;
+    } data;
+} ompi_count_array;
```

Comment:
Typedefs should end in `_t`:

```suggestion
} ompi_count_array_t;
```